src/Pate/Memory/MemTrace.hs

```diff
@@ -703,6 +715,63 @@ execMacawStmtExtension (MacawArchEvalFn archStmtFn) mkundef mvar globs stmt
       off' <- bvZext sym w offset
       pure (LLVMPointer region off')
 
+newtype MacawSyscallModel sym arch = MacawSyscallModel ()
+
+installMacawSyscallHandler ::
+  IsSymInterface sym =>
+  Assignment TypeRepr atps ->
+  Assignment TypeRepr rtps ->
+  MacawSyscallModel sym arch ->
+  GlobalVar (MemTrace arch) ->
+  CrucibleState p sym ext rtp blocks r ctx ->
+  IO (FnVal sym atps (StructType rtps), CrucibleState p sym ext rtp blocks r ctx)
+installMacawSyscallHandler atps rtps syscallModel mvar cst =
+  do let simCtx = cst^.stateContext
+     let halloc = simHandleAllocator simCtx
+     let nm = "MacawSyscall"
+     fnh <- mkHandle' halloc nm atps (StructRepr rtps)
+     let FnBindings fns = simCtx ^. functionBindings
+     let ovr  = mkOverride' nm (StructRepr rtps) $
+                  do RegMap args <- getOverrideArgs
+                     applySyscallModel rtps args syscallModel mvar
+     let fns' = FnBindings (insertHandleMap fnh (UseOverride ovr) fns)
+     return (HandleFnVal fnh, cst & stateContext . functionBindings .~ fns')
```

Comment:
In general, is it okay to re-install the same handler multiple times? I guess it is just basically the same handler under many different handles

Reply:
Right now, we're making a fresh name each time, so there's no conflict.  We could cache the handlers and reuse them, but I'm not sure it is worth the trouble.